api/src/main/java/org/apache/iceberg/encryption/EncryptionKeyMetadata.java

```diff
@@ -49,4 +49,12 @@ static EncryptionKeyMetadata empty() {
   ByteBuffer buffer();
 
   EncryptionKeyMetadata copy();
+
+  default ByteBuffer encryptionKey() {
+    return null;
```

Comment:
Why `null` instead of `UnsupportedOperationException`? Won't the caller need to throw an exception?